lib/msf/core/feature_manager.rb

```diff
@@ -0,0 +1,67 @@
+# -*- coding: binary -*-
+# frozen_string_literal: true
+
+require 'msf/core/plugin'
+
+module Msf
+  ###
+  #
+  # The feature manager is responsible for managing feature flags that can change characteristics of framework.
+  #
+  ###
+  class FeatureManager
+
+    include Framework::Offspring
+
+    WRAPPED_TABLES = 'wrapped_tables'
+    DEFAULTS = [
+      {
+        name: 'wrapped_tables',
+        description: 'When enabled Metasploit will wordwrap all tables to fit into the available terminal width',
+        enabled: false
+      }.freeze
+    ].freeze
+
+    #
+    # Initializes the feature manager.
+    #
+    def initialize(framework)
+      @framework = framework
+      @flag_lookup = DEFAULTS.each_with_object({}) do |feature, acc|
+        key = feature[:name]
+        acc[key] = feature.dup
+      end
+    end
+
+    def all
+      @flag_lookup.values
+    end
+
+    def enabled?(name)
+      return false unless @flag_lookup[name]
+      @flag_lookup[name][:enabled] == true
+    end
+
+    def exists?(name)
+      @flag_lookup.key?(name)
+    end
+
+    def names
+      all.map { |feature| feature[:name] }
+    end
+
+    def set(name, value)
+      return false unless @flag_lookup[name]
+
+      @flag_lookup[name][:enabled] = value
+
+      if name == WRAPPED_TABLES
+        if value
+          Rex::Text::Table.wrap_tables!
+        else
+          Rex::Text::Table.unwrap_tables!
+        end
+      end
```

Comment:
I imagine most feature flags to be more passive than choosing to make the feature manager have an explicit side effect. Context on this chosen approach:

https://github.com/rapid7/rex-text/pull/32/files#r452150096